demisto_sdk/commands/content_graph/interface/neo4j/queries/dependencies.py

```diff
@@ -113,25 +125,51 @@ def update_marketplaces_property(tx: Transaction, marketplace: str) -> None:
         )
-        RETURN content_item.node_id AS excluded_content_item, dependency.node_id AS reason
+        RETURN content_item.node_id AS excluded_content_item, dependency.content_type + ":" + dependency.object_id AS reason
     """
     result = run_query(tx, query)
     outputs: Dict[str, Set[str]] = {}
     for row in result:
         outputs.setdefault(row["excluded_content_item"], set()).add(row["reason"])
     logger.info(f"Removed {marketplace} from marketplaces for {len(outputs.keys())} content items.")
+    logger.debug(f"Excluded content items: {dict(sorted(outputs.items()))}")
 
 
 def update_uses_for_integration_commands(tx: Transaction) -> None:
+    """This query creates a relationships between content items and integrations, based on the commands they use.
+    If a content item uses a command which is in an integration, we create a relationship between the content item and the integration.
+    The mandatorily property is calculated as follows:
+        - If there is only one integration that implements the command, the mandatorily property is the same as the command's mandatorily property.
+          Otherwise, the mandatorily property is false.
+        - If there is already a relationship between the content item and the integration,
+          the mandatorily property is the OR of the existing and the new mandatorily property.
+
+    Args:
+        tx (Transaction): _description_
+    """
     query = f"""
-    MATCH (content_item:{ContentType.BASE_CONTENT})-[r:{RelationshipType.USES}]->(command:{ContentType.COMMAND})
-    MATCH (command)<-[rcmd:{RelationshipType.HAS_COMMAND}]-(integration:{ContentType.INTEGRATION})
+    MATCH (content_item:{ContentType.BASE_CONTENT})
+        -[r:{RelationshipType.USES}]->
+            (command:{ContentType.COMMAND})<-[rcmd:{RelationshipType.HAS_COMMAND}]
+            -(integration:{ContentType.INTEGRATION})
+    WHERE {is_target_available("content_item", "integration")}
+    AND NOT command.object_id IN {list(GENERIC_COMMANDS_NAMES)}
+    WITH command, count(DISTINCT rcmd) as command_count
+
+    MATCH (content_item:{ContentType.BASE_CONTENT})
+        -[r:{RelationshipType.USES}]->
+            (command)<-[rcmd:{RelationshipType.HAS_COMMAND}]
+            -(integration:{ContentType.INTEGRATION})
     WHERE {is_target_available("content_item", "integration")}
+    AND NOT command.object_id IN {list(GENERIC_COMMANDS_NAMES)}
 
-    WITH count(rcmd) as command_count, content_item, r, integration
     MERGE (content_item)-[u:USES]->(integration)
-    SET u.mandatorily = u.mandatorily AND (CASE WHEN command_count = 1 THEN true ELSE false END)
+    ON CREATE
+        SET u.mandatorily = CASE WHEN command_count = 1 THEN r.mandatorily ELSE false END
+    ON MATCH
+        SET u.mandatorily = u.mandatorily OR (CASE WHEN command_count = 1 THEN r.mandatorily ELSE false END)
     RETURN count(u) as uses_relationships
+
```

Comment:
let's log the query outputs (please check)
```suggestion
        SET u.mandatorily = u.mandatorily OR (CASE WHEN command_count = 1 THEN r.mandatorily ELSE false END)
    RETURN
        content_item.node_id AS content_item,
        r.mandatorily AS is_cmd_mandatory,
        collect(integration.object_id) AS integrations,
        u.mandatorily AS is_integ_mandatory,
        command.name AS command
"""
    result = run_query(tx, query)
    for row in result:
        content_item = row["content_item"]
        command = row["command"]
        is_cmd_mandatory = "mandatory" if row["is_cmd_mandatory"] else "optional"
        integrations = row["integrations"]
        is_integ_mandatory = "mandatory" if row["is_integ_mandatory"] else "optional"
        content_item = row["content_item"]
        msg = f"{content_item} uses command {command} ({is_cmd_mandatory}), "
        msg += f"new {is_integ_mandatory} rels to integrations: {integrations}"
        logger.debug(msg)


```